src/language_features/document_symbol.rs

```diff
@@ -216,14 +235,41 @@
     ctx: &Context,
 ) -> String {
     fn format_symbol_at_depth<'a, T: Symbol<T>>(
-        output: &mut Vec<(String, String, &'a str)>,
+        output: &mut Vec<(String, String, String)>,
         items: &'a [T],
         meta: &EditorMeta,
         server: &ServerSettings,
         ctx: &Context,
         depth: usize,
+        prefix: &Vec<bool>,
```

Comment:
This should mean that `depth` is redundant (it's `prefix.len()`) so we should remove it

Reply:
Done